### PR TITLE
Bats tests

### PR DESCRIPTION
**Pull request by fairclothjm**
This PR adds BATs acceptance tests for vault-plugin-database-oracle

```
$ ./test.bats
 ✓ Read license
 ✓ POST /database/config/:name - write oracle connection config
 ✓ GET /database/config/:name - read oracle connection config
 ✓ LIST /database/config - list configs
 ✓ DELETE /database/config/:name - delete oracle connection config
 ✓ POST /database/reset/:name - reset oracle connection
 ✓ POST /database/roles/:name - write role
 ✓ GET /database/roles/:name - read role
 ✓ LIST /database/roles - list roles
 ✓ DELETE /database/roles/:name - delete role
 ✓ GET /database/creds/:name - generate dynamic credentials
 ✓ Execute sqlplus query with dynamic credentials
 ✓ Execute sqlplus query with static credentials
 ✓ GET /database/static-creds/:name - read the current credentials for the static role
 ✓ POST /database/rotate-root/:name - rotate the root user credentials for the database connection

15 tests, 0 failures

```

### CODE REVIEW

File: tests/test.bats [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+    connection_url="{{username}}/{{password}}@localhost:1521/ORCLPDB1" \
+    allowed_roles="*" \
+    username="$VAULT_USER" \
+    password="$VPASS" \
+    max_conneciton_lifetime="30s"
+  assert_status 0
+}
+
+@test "GET /database/config/:name - read oracle connection config" {
+  run vault read --namespace=ns1 database/config/$DB_NAME
```
Comment (tomhjp) [inline tests/test.bats:193]: Might be nice to have some assertions that the contents are correct too? You could use `-format=json` and do some processing with `jq`.

File: tests/test.bats [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+    default_ttl="1h" \
+    max_ttl="1h"
+  assert_status 0
+
+  run vault delete --namespace=ns1 database/roles/delete-me
+  assert_status 0
+}
+
+@test "GET /database/creds/:name - generate dynamic credentials" {
+  run vault read --namespace=ns1 database/creds/$VROLE
```
Comment (tomhjp) [inline tests/test.bats:257]: Might be nice to confirm the TTL is as expected?

**Review by tomhjp — APPROVED** [review]
Looks great! Just a couple of optional suggestions for extra assertions

File: tests/test.bats [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+}
+
+@test "POST /database/config/:name - write oracle connection config" {
+  run vault write --namespace=ns1 database/config/$DB_NAME \
+    plugin_name=vault-plugin-database-oracle \
+    connection_url="{{username}}/{{password}}@localhost:1521/ORCLPDB1" \
+    allowed_roles="*" \
+    username="$VAULT_USER" \
+    password="$VPASS" \
+    max_conneciton_lifetime="30s"
```
Comment (fairclothjm) [inline tests/test.bats:188]:
```suggestion
    max_connection_lifetime="30s"
```

File: tests/test.bats (hunk 1 above)
Comment (fairclothjm) [reply tests/test.bats:193]: will do thanks!

File: tests/test.bats (hunk 2 above)
Comment (fairclothjm) [reply tests/test.bats:257]: good idea!

**Review by tomhjp — APPROVED** [review]

**Review by calvn — APPROVED** [review]

**Pull request merged**